pkg/controllers/crds/crds_controller.go

```diff
@@ -72,13 +72,15 @@ type Reconciler struct {
 	RequeueInterval time.Duration
 
 	// the controller is ready when all crds are injected
+	// and the controller is elected as leader
+	isLeader       bool
```

Comment:
Chane this to `<-chan struct`.

Then in the leader function simply check on it:

```go
	if !r.ready {
		select {
		case <-r.isLeader:
			r.ready = true
		default:
			return nil
		}
	}
```

This way, you're not blocking the `ReadyCheck` and you can always read from a closed channel. That `<-r.isLeader` will always call true when the channel gets closed. We don't care if it's a closed read since the channel is closed when the leader is elected.